var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -68,6 +68,10 @@ class Kokkos(Package):
         description='Set the GPU architecture to use'
     )
 
+    # conflicts
+    conflicts('~cuda', when='gpu_arch!=None',
```

Comment:
Spack's spec syntax doesn't have a `!=` unfortunately. @alalazo is there any easy way to check this or should we just keep the check in `install`?

Reply:
There is an example of iterating through a list in the adios package that might be adaptable as well:
```for p in ['+hdf5', '+netcdf', 'staging=flexpath', 'staging=dataspaces']:
        conflicts(p, when='~mpi')```

Reply:
It would be verbose, though, if it had to specify all of the gpu_arch values.

Reply:
Yeah, that's basically what you would have to do. You could store the list of `gpu_arch` values in a list and use it both in the `variant` definition and the `conflicts` directive.